### PR TITLE
kube-state-metrics based scalability rules + default metrics config

### PR DESCRIPTION
**Pull request by mikouaj**
* rewritten existing scalability rules to use new metrics + introduced new rules
* added default set of metrics into the code
* fixed validation and default functions for "check scalability" command

### CODE REVIEW

File: gke-policies-v2/scalability/limits_nodes_per_pool_zone_test.rego [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package gke.scalability.nodes_per_pool_zone
+
+test_nodes_per_pool_zone_above_warn_limit {
+	not valid with input as {"data": {"monitoring": {"nodes_per_pool_zone": { "name": "nodes_per_pool_zone", "vector": {"default-pool": {"europe-central2-a": 642, "europe-central2-b": 734,"europe-central2-a": 821}}}}}}
```
Comment (ewojtach) [inline gke-policies-v2/scalability/limits_nodes_per_pool_zone_test.rego:18]: shouldn't we add to the test structure autopilot.enabled information? as it is checked in the rule

File: gke-policies-v2/scalability/limits_pods.rego [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    input.data.monitoring.pods.scalar > warn_limit
+	msg := sprintf("Total number of pods %d has reached warning level %d (limit is %d)", [input.data.monitoring.pods.scalar, warn_limit, limit_standard])
+}
+
+violation[msg] {
+	warn_limit = round(limit_autopilot * threshold * 0.01)
+	input.data.gke.autopilot
+	input.data.gke.autopilot.enabled
+    input.data.monitoring.pods.scalar > warn_limit
+	msg := sprintf("Total number of pods %d has reached warning level %d (limit is %d)", [input.data.monitoring.pods.scalar, warn_limit, limit_autopilot])
```
Comment (ewojtach) [inline gke-policies-v2/scalability/limits_pods.rego:51]: we could add here information that this limit is for the autopilot ("limit is %d for autopilot cluster")

File: gke-policies-v2/scalability/limits_pods_per_node.rego [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# METADATA
+# title: Number of PODs per node
+# description: The total number of PODs running on a single node
+# custom:
+#   group: Scalability
+#   severity: Low
+#   recommendation: >
+#     When the limit is reached, the kube-scheduler will assign the new PODs to the other nodes, if available.
+#     If there will be no nodes available to assign the POD, the cluster autoscaler, if configured,  may add new nodes.
```
Comment (ewojtach) [inline gke-policies-v2/scalability/limits_pods_per_node.rego:23]: additional space between "if configured, " and "may add new nodes" ;-)

**Review by ewojtach — APPROVED** [review]
great job! LGTM!
added few minor comments (mainly for typos)

File: gke-policies-v2/scalability/limit_secrets_encryption_test.rego [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package gke.scalability.secrets_with_enc
+
+test_secrets_with_enc_above_warn_limit {
+	not valid with input as {"data": {"monitoring": {"secrets": { "name": "secrets", "scalar": 28000}}, "gke": {"name": "cluster-1", "database_encryption": {"state": 1}}}}
+}
+
+test_secrets_with_enc_within_warn_limit {
```
[the remark this answers is not in the document]
Comment (mikouaj) [reply gke-policies-v2/scalability/limit_secrets_encryption_test.rego:21]: 👍 I've changed "wihin" to "below" in all tests

File: gke-policies-v2/scalability/limits_pods.rego (hunk 2 above)
Comment (mikouaj) [reply gke-policies-v2/scalability/limits_pods.rego]: 👍

File: gke-policies-v2/scalability/limits_pods_per_node.rego (hunk 3 above)
Comment (mikouaj) [reply gke-policies-v2/scalability/limits_pods_per_node.rego]: 👍

File: gke-policies-v2/scalability/limits_nodes_per_pool_zone_test.rego (hunk 1 above)
Comment (mikouaj) [reply gke-policies-v2/scalability/limits_nodes_per_pool_zone_test.rego:18]: Makes total sense, thank you

File: gke-policies-v2/scalability/limits_containers.rego [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	not input.data.gke.autopilot.enabled
+    input.data.monitoring.containers.scalar > warn_limit
+	msg := sprintf("Total number of containers %d has reached warning level %d (limit is %d)", [input.data.monitoring.containers.scalar, warn_limit, limit_standard])
+}
+
+violation[msg] {
+	warn_limit = round(limit_autopilot * threshold * 0.01)
+	input.data.gke.autopilot.enabled
+    input.data.monitoring.containers.scalar > warn_limit
+	msg := sprintf("Total number of containers %d has reached warning level %d (limit is %d)", [input.data.monitoring.containers.scalar, warn_limit, limit_autopilot])
```
[the remark this answers is not in the document]
Comment (mikouaj) [reply gke-policies-v2/scalability/limits_containers.rego]: 👍

**Pull request merged**